db/transaction_log_impl.cc

```diff
@@ -276,12 +289,28 @@ void TransactionLogIteratorImpl::UpdateCurrentWriteBatch(const Slice& record) {
   // currentBatchSeq_ can only change here
   assert(current_last_seq_ <= versions_->LastSequence());
 
+  if (read_options_.with_cache_) {
+    // cache the mapping of sequence to log block index when seeking to the
+    // start or end sequence
+    if ((current_batch_seq_ <= starting_sequence_number_ &&
+         current_last_seq_ >= starting_sequence_number_) ||
+        current_last_seq_ == versions_->LastSequence()) {
+      transaction_log_seq_cache_->Insert(
```

Comment:
This part I think will need to check if the log has compression enabled or not, before adding it to cache. Even though this session's DBOptions disables WAL compression. `getUpdatesSince` may need to read WAL files generated from previous sessions that could have WAL compression enabled. 

If it could happen that only the results for a subset of the log files can be cached, what is the impact on the correctness of this caching. I haven't think through this, can you show if it will continue to work on not work anymore.  We can maybe come up with solution based on that.